src/Assets/validationFunctions.tsx

```diff
@@ -205,10 +205,75 @@ const displayHouseholdMemberAgeHelperText: MessageFunction<string> = (applicantA
   }
 };
 
+const date = new Date();
+const CURRENT_YEAR = date.getFullYear();
+// the getMonth method returns January as 0
+const CURRENT_MONTH = date.getMonth() + 1;
+
+function birthMonthHasError(birthMonth: number | null) {
+  if (birthMonth === null) {
+    return true;
+  }
+
+  if (birthMonth < 1 || birthMonth > 12) {
+    return true;
+  }
+
+  return false;
+}
+
+function birthMonthErrorMessage() {
+  return (
+    <ErrorMessageWrapper fontSize="1rem">
+      <FormattedMessage id="ageInput.month.error" defaultMessage="Please enter a birth month." />
+    </ErrorMessageWrapper>
+  );
+}
+
+const MAX_AGE = 130;
+function birthYearHasError(birthYear: number | null) {
+  if (birthYear === null) {
+    return true;
+  }
+
+  if (birthYear > CURRENT_YEAR || birthYear < CURRENT_YEAR - MAX_AGE) {
+    return true;
+  }
+
+  return false;
+}
+
+function birthYearErrorMessage() {
+  return (
+    <ErrorMessageWrapper fontSize="1rem">
+      <FormattedMessage id="ageInput.year.error" defaultMessage="Please enter a birth year." />
+    </ErrorMessageWrapper>
+  );
+}
+
+function birthMonthInvalidWithYearErrorMessage() {
+  return (
+    <ErrorMessageWrapper fontSize="1rem">
+      <FormattedMessage id="ageInput.year.error" defaultMessage="This birth month is in the future" />
+    </ErrorMessageWrapper>
+  );
+}
+
+function birthMonthInvalidWithYearError({ birthYear, birthMonth }: { birthYear: number; birthMonth: number }) {
+  return birthYear === CURRENT_YEAR && birthMonth > CURRENT_MONTH;
+}
+
 const personDataIsValid: ValidationFunction<HouseholdData> = (householdDataState) => {
-  const { age, relationshipToHH, hasIncome, incomeStreams, healthInsurance } = householdDataState;
+  const { birthYear, birthMonth, relationshipToHH, hasIncome, incomeStreams, healthInsurance } = householdDataState;
+
+  if (birthMonth === undefined || birthYear === undefined) {
+    return false;
+  }
 
-  const ageIsValid = Number(age) >= 0 && age !== '';
+  const ageIsValid =
+    !birthYearHasError(birthYear) &&
+    !birthMonthHasError(birthMonth) &&
+    !birthMonthInvalidWithYearError({ birthYear, birthMonth });
```

Comment:
What do you think about refactoring these functions to be `birthYearHasError` => `birthYearIsValid`, `birthMonthHasError` => `birthMonthIsValid` etc. for readability?

Reply:
All of the other validation functions are names "hasError"